rmgpy/tools/uncertainty.py

```diff
@@ -341,37 +326,65 @@ def loadModel(self, chemkinPath, dictionaryPath, transportPath=None):
                                                               dictionaryPath=dictionaryPath,
                                                               transportPath=transportPath)
 
+    def retrieveSaturatedSpeciesFromList(self,species):
+        """
+        Given a radical `species`, this function retrieves the saturated species objects from a list of species objects.
+        """
+        
+        molecule = species.molecule[0]
+        assert molecule.isRadical(), "Method only valid for radicals."
+        saturatedStruct = molecule.copy(deep=True)
+        saturatedStruct.saturate()
+        for otherSpecies in self.speciesList:
+            if otherSpecies.isIsomorphic(saturatedStruct):
+                return otherSpecies, False
+        else:
+            for key,value in self.database.thermo.libraries.iteritems():
+                for entry in value.entries.values():
+                    if entry.item.isIsomorphic(saturatedStruct):
+                        newSpc = Species().fromAdjacencyList(saturatedStruct.toAdjacencyList())
+                        newSpc.thermo = entry.data
+                        self.speciesList.append(newSpc)
+                        return newSpc, True
```

Comment:
If I understand this correctly, this function is used to return the species object of the saturated molecule in the model to use its sources of thermo estimation to later determine the uncertainty. As you noted, it is possible for a radical species to be present in the model without the corresponding saturated species being present. The idea of your code here is that it is a fall-back for this case, and searches our libraries to see if the molecule exists there.

Specifically, it looks like the uncertainty module loads whichever thermo libraries the user specifies, so as long as the user specifies the same libraries in the same order as they did when they made the RMG model. Assuming the user does this, and we are still unable to find the species in the libraries, could we not assume that the saturated species was estimated by group additivity? If so we should implement a final fall-back method that returns a species object with thermo estimated with group additivity.

tldr: Is a fall-back method for group-additivity needed?

Reply:
In `extractSourceFromComments`, HBI and GAV are considered a single source. Retrieving the saturated structure is only done if the thermo was from two sources, ie. HBI on top of a library value or HBI on top of a QM value.

Reply:
Gotcha, so a fall-back to GAV is not only unnecessary but would give the wrong result (since we only would get here if GAV was not used)?

Reply:
I think so, if I understood correctly.